File: data/problems copie/1/result/1.output_3.py
# @lc app=leetcode id=1 lang=python3
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement_dict = {}
        
        for i, num in enumerate(nums):
            complement = target - num
            if complement in complement_dict:
                return [complement_dict[complement], i]
            
            complement_dict[num] = i
        
        return []

File: data/EXP01/golang/1/1.output_3.go
func twoSum(nums []int, target int) []int {
    complement := make(map[int]int)
    
    for i, num := range nums {
        complementNum := target - num
        
        if j, ok := complement[complementNum]; ok {
            return []int{j, i}
        }
        
        complement[num] = i
    }
    
    return []int{}
}


    target := 9
    result := twoSum(nums, target)
    fmt.Println(result) // Output: [0 1]
}
